### PR TITLE
docs: Counterfactual prediction user guide

### PR DESCRIPTION
**Pull request by justjhong**
Addresses #1176

### CODE REVIEW

File: docs/user_guide/background/counterfactual_prediction.md [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+Suppose we have a trained model $f_\theta$ that takes in a data point $x$ (e.g., gene expression counts) and a condition $c$ (e.g., treatment group) and returns a prediction $\hat{y}$.
+Each data point takes the form of a tuple $(x,c) \in \mathcal{D}$.
+We can define a *counterfactual query* as a pair $(x,c')$ where $c' \neq c$,
+and the respective model output as the *counterfactual prediction*, $\hat{y}' = f_\theta(x,c')$.
+
+We separate $c$ here out from $x$ to make the counterfactual portion of the query explicit, but it can be thought of as another dimension of $x$.
+
+## In-distribution vs. out-of-distribution
+
+Since we are working with statistical models rather than causal models, we have to careful when we can rely on counterfactual predictions. At a high level, if we assume the true function relating the features to the target is smooth, we can trust counterfactual predictions for queries that are similar to points in the training data.
```
Comment (canergen) [inline docs/user_guide/background/counterfactual_prediction.md:28]: be careful

File: docs/user_guide/background/counterfactual_prediction.md [hunk 2]
```diff
@@ -1,5 +1,61 @@
...
+
+Say we have a counterfactual query $(x,c')$.
+If we have data points in the training set $(x',c')$ that are similar to $(x,c')$,
```
Comment (canergen) [inline docs/user_guide/background/counterfactual_prediction.md:31]: and our model is reasonable well in generating y

**Comment by canergen** [comment]
Two small comments. Do you have a suggestion how to review md files after rendering?

**Comment by justjhong** [comment]
@canergen thanks for looking. To see the generated docs you can click the link generated by ci/cd

**Comment by canergen** [comment]
Thanks. Rendering looks good. Can be merged after fixing the small two comments.

**Pull request merged**